### PR TITLE
Fix local s3 target

### PR DESCRIPTION
**Pull request by kaloramik**
the local s3 path was failing when there were conflicts with s3, need to override exists function as well

### CODE REVIEW

File: fireflower/targets.py [hunk 1]
```diff
@@ -28,19 +28,33 @@ class FireflowerS3Target(S3Target):
 
     fs = None
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.local_s3_path = os.getenv('LOCAL_S3_PATH', None)
```
Comment (ihat) [inline fireflower/targets.py]: Stylistically, would extract `os.getenv('LOCAL_S3_PATH', None)` into class constant, and
`self.local_s3_path = kwargs.get('LOCAL_S3_PATH', LOCAL_S3_PATH)`

File: fireflower/targets.py [hunk 2]
```diff
@@ -28,19 +28,33 @@ class FireflowerS3Target(S3Target):
...
-        if not local_s3_path:
+        if not self.local_s3_path:
             return super().open(mode)
 
-        modified_path = self.path.replace('s3://', '')
-        new_path = os.path.join(local_s3_path, modified_path)
-
         is_compressed = getattr(self, 'compressed', False)
 
+        new_path = self.local_path
```
Comment (ihat) [inline fireflower/targets.py]: Would inline `self.local_path`

File: tests/targets_tests.py [hunk 3]
```diff
@@ -164,3 +165,33 @@ def test_s3_typed_uncompressed_csv_target(self):
         s.write_csv(df, index=False)
         read_result = s.read_csv()
         self.assertDictEqual(df.to_dict(), read_result.to_dict())
+
+    @mock_s3
+    def test_local_exists(self):
+        with TempDirectory() as d:
```
Comment (ihat) [inline tests/targets_tests.py]: Can you add a comment here explaining that "local_exists returns True even if target exists in S3"?

File: tests/targets_tests.py [hunk 4]
```diff
@@ -164,3 +165,33 @@ def test_s3_typed_uncompressed_csv_target(self):
...
+                self.assertEqual(v, 'LOCAL_S3_PATH')
+                return d.path
+
+            with mock.patch('fireflower.targets.os.getenv',
+                            side_effect=getenv):
+                # even though it exists on s3, check that it doesnt exist locally
+                t = S3CSVTarget(dest_path, compressed=False)
+                self.assertFalse(t.exists())
+
+                os.mkdir(os.path.join(d.path, 'some_bucket'))
```
Comment (ihat) [inline tests/targets_tests.py]: use bucket_name here.

File: tests/targets_tests.py [hunk 5]
```diff
@@ -164,3 +165,33 @@ def test_s3_typed_uncompressed_csv_target(self):
...
+
+    @mock_s3
+    def test_local_exists(self):
+        with TempDirectory() as d:
+            conn = boto.connect_s3()
+            bucket_name = 'some_bucket'
+            file_name = 'some_file.csv.gz'
+            dest_path = 's3://%s/%s' % (bucket_name, file_name)
+            conn.create_bucket(bucket_name)
+            s = S3CSVTarget(dest_path, compressed=False)
```
Comment (ihat) [inline tests/targets_tests.py]: nit: order of df and s. Swap line 177 with 178.

**Comment by ihat** [comment]
Looks good.

File: fireflower/targets.py (hunk 1 above)
Comment (kaloramik) [inline fireflower/targets.py]: actually, making this a class variable makes it hard to test, as the class will then read in the variable, before mocking.

**Pull request merged**